app/lib/activitypub/activity.rb

```diff
@@ -176,4 +178,33 @@ def reject_payload!
     Rails.logger.info("Rejected #{@json['type']} activity #{@json['id']} from #{@account.uri}#{@options[:relayed_through_actor] && "via #{@options[:relayed_through_actor].uri}"}")
     nil
   end
+
+  # Ensure all emojis declared in the activity's tags are
+  # present in the database and downloaded to the local cache.
+  # Required by EmojiReact and Like for emoji reactions.
+  def process_emoji_tags(tags)
+    as_array(tags).each do |tag|
+      process_single_emoji tag if tag['type'] == 'Emoji'
+    end
+  end
```

Comment:
There is no reason to do this.
Both Misskeys `Like` activity as well as the `EmojiReact` activity only expect a single tag object.